squad_client/core/models.py

```diff
@@ -87,6 +87,15 @@ def get(self, _id):
 
 class Squad(SquadObject):
 
+    def fetch_id(self, klass, id=None):
```

Comment:
What if instead of having `fetch_id`, we had a constructor method in `SquadObject` that accepts the object id, then we call up the method `__fetch__` which will do the fetching and error validation for you. Does that make sense?

Reply:
doesn't this mean we have to provide a constructor to each subclass since they all subclass SquadObject?